app/src/ui/discard-changes/discard-changes-dialog.tsx

```diff
@@ -57,19 +62,37 @@ export class DiscardChanges extends React.Component<
   }
 
   private getOkButtonLabel() {
-    if (this.props.discardingAllChanges) {
-      return __DARWIN__ ? 'Discard All Changes' : 'Discard all changes'
+    switch (this.props.discardType) {
+      case DiscardType.AllFiles:
+        return __DARWIN__ ? 'Discard All Changes' : 'Discard all changes'
+      case DiscardType.SomeFiles:
+      case DiscardType.Selection:
+        return __DARWIN__ ? 'Discard Changes' : 'Discard changes'
+      default:
+        return assertNever(
+          this.props.discardType,
+          'Invalid discardType property'
+        )
     }
-    return __DARWIN__ ? 'Discard changes' : 'Discard changes'
   }
 
   private getDialogTitle() {
-    if (this.props.discardingAllChanges) {
-      return __DARWIN__
-        ? 'Confirm Discard All Changes'
-        : 'Confirm Discard all changes'
+    switch (this.props.discardType) {
```

Comment:
If we grab this out of  props we can get the `assertNever` to fit nicely onto one line

```suggestion
    const { discardType } = this.props

    switch (discardType) {
```